src/python/pants/backend/python/subsystems/pex_bootstrap/_ptex_launcher.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# This bootstrapping technique for lazily loading requirements comes from the pex project test case:
+# https://github.com/pantsbuild/pex/commit/66aa5ef69e8b65004a548b44edc0b81dfa8ec736#diff-b5b57c9d4fc663f59ecaedb851c13bd8R23
+
+import json
+import os
+import sys
+
+from pex import resolver
+from pex.common import open_zip
+from pex.fetcher import Fetcher, PyPIFetcher
+from pex.interpreter import PythonInterpreter
+from pex.pex_builder import PEXBuilder
+from pex.pex_info import PexInfo
+from pex.platforms import Platform
+
+
+self = sys.argv[0]
+filename_base, ext = tuple(os.path.splitext(self))
+hydrated_ipex_file = '{}.ipex'.format(filename_base)
+
+if not os.path.isfile(hydrated_ipex_file):
+  print('Hydrating {} to {}'.format(self, hydrated_ipex_file))
+
+  wrapper_ptex_info = PexInfo.from_pex(self)
+  if wrapper_ptex_info.zip_safe:
+    raise ValueError("Could not hydrate ptex: ipex info must be marked zip_safe=False. This ipex file's info was:\\n{}".format(wrapper_ptex_info.dump(indent=4)))
+  code_root = os.path.join(wrapper_ptex_info.zip_unsafe_cache, wrapper_ptex_info.code_hash)
+  with open_zip(self) as zf:
+    # Populate the pex with the pinned requirements and distribution names & hashes.
+    ipex_info = PexInfo.from_json(zf.read('IPEX-INFO'))
+    ipex_builder = PEXBuilder(pex_info=ipex_info)
+
+    # Populate the pex with the needed code.
+    ptex_info = json.loads(zf.read('PTEX-INFO').decode('utf-8'))
+    for path in ptex_info['code']:
+      ipex_builder.add_source(os.path.join(code_root, path), path)
```

Comment:
You already loaded the sourcees and resources into the ptex here: https://github.com/pantsbuild/pants/pull/8793/files#diff-5f2c007abfa9fbf0323826392a6374c7R373

Instead of digging into pex internals (code_root), just use the already open `zf` of yourself (ptex), to grab the sources.